language/move-vm/transactional-tests/tests/instructions/arithmetic_operators_u32.mvir

```diff
@@ -0,0 +1,164 @@
+//# run
+main() {
+label b0:
+    assert(0u32 + 0u32 == 0u32, 1000);
+    assert(0u32 + 1u32 == 1u32, 1001);
+    assert(1u32 + 1u32 == 2u32, 1002);
+
+    assert(13u32 + 67u32 == 80u32, 1100);
+    assert(100u32 + 10u32 == 110u32, 1101);
+
+    assert(0u32 + 4294967295u32 == 4294967295u32, 1200);
+    assert(1u32 + 4294967294u32 == 4294967295u32, 1201);
+    assert(5u32 + 4294967290u32 == 4294967295u32, 1202);
+    return;
+}
+
+//# run
+main() {
+label b0:
+    _ = 1u32 + 4294967295u32;
+    return;
+}
+
+//# run
+main() {
+label b0:
+    _ = 4294000000u32 + 4294000000u32;
+    return;
+}
+
+
+
+//# run
+main() {
+label b0:
+    assert(0u32 - 0u32 == 0u32, 2000);
+    assert(1u32 - 0u32 == 1u32, 2001);
+    assert(1u32 - 1u32 == 0u32, 2002);
+
+    assert(52u32 - 13u32 == 39u32, 2100);
+    assert(100u32 - 10u32 == 90u32, 2101);
+
+    assert(4294967295u32 - 4294967295u32 == 0u32, 2200);
+    assert(5u32 - 1u32 - 4u32 == 0u32, 2201);
+    return;
+}
+
+//# run
+main() {
+label b0:
+    _ = 0u32 - 1u32;
+    return;
+}
+
+//# run
+main() {
+label b0:
+    _ = 54u32 - 100u32;
+    return;
+}
+
+
+//# run
+main() {
+label b0:
+    assert(0u32 * 0u32 == 0u32, 3000);
+    assert(1u32 * 0u32 == 0u32, 3001);
+    assert(1u32 * 1u32 == 1u32, 3002);
+
+    assert(6u32 * 7u32 == 42u32, 3100);
+    assert(10u32 * 10u32 == 100u32, 3101);
+
+    assert(2147483647u32 * 2u32 == 4294967294u32, 3200);
+    return;
+}
+
+//# run
+main() {
+label b0:
+    _ = 429496u32 * 429496u32;
+    return;
+}
+
+//# run
+main() {
+label b0:
+    _ = 2147483647 * 2u32;
```

Comment:
Is this supposed to be a type error? If so, we should move it to the bytecode verifier.

Reply:
@tnowacki fixed. Last digit should've been 8, and suffixed. Was supposed to be arithmetics tests